.github/workflows/auto-release-bugfix.yml

```diff
@@ -0,0 +1,173 @@
+                  # Redirecting stderr to /dev/null to keep actions logs
+                  # reflecting the expected state of the program.
+                  if hub ci-status $SOURCE_BRANCH 2> /dev/null
+                  then
+                      echo "There are test failures on $SOURCE_BRANCH; aborting release."
+                      exit 1
+                  fi
+
+                  # Don't do a bugfix release if target branch's HEAD already has a tag
+                  # git-describe exits 0 if a tag is found, 128 if not.
+                  # Redirecting stderr to /dev/null to keep the expected
+                  # 'fatal' error message out of the actions logs.
+                  if git describe --exact-match --tags $SOURCE_BRANCH 2> /dev/null
+                  then
+                      echo "The latest commit on $SOURCE_BRANCH already has a tag"
+                  else
+                      LATEST_TAG=$(git describe --tags | awk -F '-' '{print $1}')
+                      echo "Latest tag on $SOURCE_BRANCH is $LATEST_TAG"
+                      if [[ $LATEST_TAG =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]
+                      then
+                          echo "$LATEST_TAG identified as a full release."
+                          echo "Proceeding with automated bugfix release."
+                      else
+                          echo "$LATEST_TAG does not conform to bugfix versioning."
+                          echo "This script does not know how to handle non-bugfix releases."
+                          echo "Aborting."
+                          exit 1
+                      fi
+
+                      # Exporting these variables is necessary for envsubst.
+                      export BUGFIX_VERSION=$(python setup.py --version | awk -F '.' '{print $1"."$2"."$3+1}')
+                      export TARGET_BRANCH=autorelease/$BUGFIX_VERSION
+
+                      # If $TARGET_BRANCH already exists, that means there's already
+                      # a PR open for it and we shouldn't do another release.
+                      if git ls-remote --exit-code --heads https://github.com/$GITHUB_REPOSITORY.git $TARGET_BRANCH
+                      then
+                          echo "The branch $TARGET_BRANCH already exists."
+                          echo "This is most likely because there's an "
+                          echo "outstanding PR for it."
+                          echo ""
+                          echo "Exiting the release process."
+                          exit 0
+                      fi
+
+                      # Update the UG hash in the Makefile
+                      # This could be done with curl and sed, but it'll be easier
+                      # to read this way.
+                      python ci/release/increment-userguide-revision.py
+
+                      # Update HISTORY with the tag and today's date.
+                      python ci/release/update-history.py $BUGFIX_VERSION $(date "+%Y-%m-%d")
```

Comment:
I'm not so experienced yet with Actions. Does this whole job stop if there's an exception in one of these python scripts? I think we would want it to.

Reply:
Yes, any nonzero exit code will cause the step to immediately abort with an error unless `bash` with `set -e` is used (which we aren't using).